app/api/log.py

```diff
@@ -21,3 +28,17 @@ def log_event(event, user, room=None, data=None):
     db.session.add(log)
     db.session.commit()
     return log
+
+@socketio.on('log')
+@login_required
+def log(data):
+    sender = current_user if 'sender_id' not in data else User.query.get(data['sender_id'])
+    if not sender:
+        return False, "sender not found"
+
+    if 'room' in data:
+        room = Room.query.get(data['room'])
+        if not room:
+            return False, "room not found"
+
+    log_event(data["type"], current_user, room=room, data={'message':data['message']})
```

Comment:
Missing newline at end of file
```suggestion
    log_event(data["type"], current_user, room=room, data={'message':data['message']})
    
```

Reply:
Instead of log `message`, would it make more sense to log the full data dictionary (excluding `sender_id`, `type`, and `room`)?